File: src/main/java/itmo/is/project/dto/module/storage/StoredResourceDto.java
package itmo.is.project.dto.module.storage;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import itmo.is.project.dto.resource.ResourceAmountDto;

public record StoredResourceDto(

        @Schema(example = "1")
        @JsonProperty(value = "storageModuleId", required = true)
        Integer storageModuleId,

        @JsonProperty(value = "resourceAmount", required = true)
        ResourceAmountDto resourceAmount
) {
}

File: src/main/kotlin/itmo/isproject/storage/dto/StoredResourceDto.kt
package itmo.isproject.storage.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import itmo.isproject.shared.resource.dto.ResourceAmountDto

data class StoredResourceDto(

    @Schema(example = "1")
    @JsonProperty(value = "storageModuleId", required = true)
    val storageModuleId: Int?,

    @JsonProperty(value = "resourceAmount", required = true)
    val resourceAmount: ResourceAmountDto
)
